### PR TITLE
Combine all workflow metrics into single summary

### PR DESCRIPTION
**Pull request by CliveUnger**
Combine sitrep, status, and metric_summary for each test into a single summary json at the end.

The goal is capturing a complete picture of a Workflow Run: each build success/fail, each test success/fail, the metrics for each test and also metadata like what image the tests ran on and such. This data should be all pulled from a single artifact file.

### CODE REVIEW

**Comment by CliveUnger** [comment]
I have an example in sandbox of this working here: https://github.com/NVIDIA/JAX-Toolbox/actions/runs/8461564960

There are some caveats: 
- The `metrics_summary.json` isn't included since in the sandbox I am reusing artifacts from an existing worfklow.
- The artifact path was hardcoded. This would be fixed before merging.

**Review by yhtang — CHANGES_REQUESTED** [review]

File: .github/workflows/_finalize.yaml [hunk 1]
```diff
@@ -136,39 +136,53 @@ jobs:
...
-            else
-              value=$(cat "$FILE" |  jq -r '.summary' )
+          # Combine all json files into a final summary json
+          output="combined.json"
+          combinedJson="{}"
+
+          # Loop over subdirectories in the current directory
+          for dir in all-artifacts/*; do
+            echo $dir
+            dirName=$(basename $dir) && [ -d "$dir" ] || continue
```
[the remark this answers is not in the document]
Comment (CliveUnger) [reply .github/workflows/_finalize.yaml:149]: Fixed.

**Comment by CliveUnger** [comment]
Do I need to revert the changes to `_sandbox.yaml` before this goes in?

**Comment by CliveUnger** [comment]
I am running a full "CI" workflow on this change here: https://github.com/NVIDIA/JAX-Toolbox/actions/runs/8620819287

If this completes successfully, this PR should be good to go.

**Comment by CliveUnger** [comment]
Closing, will re-open from repo branch.

**Pull request closed without merging**